### PR TITLE
Update operator infrastructure and change use statements for CMA operators (for #1738)

### PR DESCRIPTION
**Pull request by TeranIvy**
- Updates CMA operator data (`columnwise_matrix`) to `r_solver` in test kernels and examples.
- Updates operator and CMA operator infrastructure and separates them.
- Removes the use of `operator_mod` for CMA operator and operator proxy types and adds the use of `columnwise_operator_mod`.

### CODE REVIEW

**Review by arporter — CHANGES_REQUESTED** [review]
Great job Iva, I only have one small request.
All examples, tests and tutorials are OK with compilation.
Updated UG builds fine.
Does this branch work OK with LFRic?

File: examples/lfric/eg7/alg.x90 [hunk 1]
```diff
@@ -45,11 +45,12 @@ module alg
   use columnwise_op_app_kernel_mod,      only : columnwise_op_app_kernel_type
   use columnwise_op_mul_kernel_mod,      only : columnwise_op_mul_kernel_type
   use matrix_vector_kernel_mod,          only : matrix_vector_kernel_type
+  use columnwise_operator_mod,           only : columnwise_operator_type
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply examples/lfric/eg7/alg.x90:51]: You're right. Fixed.

**Comment by arporter** [comment]
Assuming the CI is green, is this ready for review again @TeranIvy?

**Comment by TeranIvy** [comment]
I am just testing this with LFRic, @arporter, so I will return it as soon as the test suite is done.

**Comment by TeranIvy** [comment]
I built the PSyclone test environment from this branch on Spice and Crays XC40 and XCS. I then ran an LFRic test branch that has all the changes for PSyclone 2.3.0 implemented (see [LFRic ticket 3288, comment 6](https://code.metoffice.gov.uk/trac/lfric/ticket/3288#comment:6)). The developer test suites were all fine. 

This PR ready to go so I am returning it to @arporter for another look.

**Review by arporter — APPROVED** [review]
All requested changes have been made.
Tests and examples are still OK with compilation.
Proceeding to merge.

**Pull request merged**